public/app/common/helpers/device/test/device-helpers.spec.js

```diff
@@ -1,17 +1,74 @@
+var _ = require('lodash');
 var DeviceHelpers = require('../');
+var mock = require('./device-helpers.mock');
 
 describe('DeviceHelpers', function () {
 
-  describe('.augmentDevice()', function () {
+  var device = mock.extractProperties(mock.device);
+  var nestWebservice = mock.extractProperties(mock.nestWebservice);
+
+  describe('augmentDevice', function () {
+    it('extracts device properties, actions and url', function () {
+      var augmentedDevice = DeviceHelpers.augmentDevice(device);
+      var alterEgoComponent = DeviceHelpers.getAlterEgoComponent(device);
+      expect(augmentedDevice.properties).toEqual(device.properties);
+      expect(augmentedDevice.actions).toEqual(_.indexBy(alterEgoComponent.actions, 'name'));
+      expect(augmentedDevice.url).toEqual(DeviceHelpers.getDeviceUrl(device));
+    });
+
+    it('includes the alter ego component and primary capability', function () {
+      var augmentedDevice = DeviceHelpers.augmentDevice(device);
+      var alterEgoComponent = DeviceHelpers.getAlterEgoComponent(device);
+      expect(augmentedDevice.alterEgoComponent).toEqual(alterEgoComponent);
+      expect(augmentedDevice.primaryCapability).toEqual(DeviceHelpers.getPrimaryCapability(alterEgoComponent));
+    });
+
     it('flags nest webservice as hidden', function () {
-      var nestWebservice = {
-        properties: { name: 'nest account', displayType: 'nest' },
-        actions: [],
-        components: [ { properties: { type: 'nest', capabilities: [ { } ] } } ],
-        links: [{ rel: ['self'] }]
-      };
-      var device = DeviceHelpers.augmentDevice(nestWebservice);
-      expect(device.hidden).toBeTruthy();
+      var nestService = DeviceHelpers.augmentDevice(nestWebservice);
+      expect(nestService.hidden).toEqual(true);
+      var augmentedDevice = DeviceHelpers.augmentDevice(device);
+      expect(augmentedDevice.hidden).toEqual(false);
```

Comment:
The second `expect` should be another test case named _'does not flag normal devices as hidden'_